Core/src/main/java/com/plotsquared/core/events/TeleportCause.java

```diff
@@ -25,8 +25,39 @@
  */
 package com.plotsquared.core.events;
 
+import java.util.EnumSet;
+
+/**
+ * The reason for an internal player teleport.
+ */
 public enum TeleportCause {
+
     COMMAND,
+    COMMAND_AREA_CREATE,
+    COMMAND_AREA_TELEPORT,
+    COMMAND_AUTO,
+    COMMAND_CLAIM,
+    COMMAND_CLEAR,
+    COMMAND_CLUSTER_TELEPORT,
+    COMMAND_DELETE,
+    COMMAND_HOME,
+    COMMAND_LIKE,
+    COMMAND_MIDDLE,
+    COMMAND_RATE,
+    COMMAND_SETUP,
+    COMMAND_TEMPLATE,
+    COMMAND_VISIT,
+    DEATH,
+    DENIED,
+    KICK,
+    LOGIN,
     PLUGIN,
-    UNKNOWN
+    UNKNOWN;
+
+    public static final class SETS {
+
+        public static final EnumSet<TeleportCause> COMMAND = EnumSet.range(TeleportCause.COMMAND, TeleportCause.COMMAND_VISIT);
```

Comment:
While I doubt anyone will mess with that, the sets should probably be wrapped into unmodifiable sets (and therefore also change the left hand side type)